proto/api_v2/model.proto

```diff
@@ -171,3 +171,648 @@ message DependencyLink {
+// summary statistics for a population of values, it may optionally contain the
+// distribution of those values across a set of buckets.
+message HistogramDataPoint {
+  // The set of labels that uniquely identify this timeseries.
+  repeated StringKeyValue labels = 1;
+
+  // start_time_unix_nano is the last time when the aggregation value was reset
+  // to "zero". For some metric types this is ignored, see data types for more
+  // details.
+  //
+  // The aggregation value is over the time interval (start_time_unix_nano,
+  // time_unix_nano].
+  //
+  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January
+  // 1970.
+  //
+  // Value of 0 indicates that the timestamp is unspecified. In that case the
+  // timestamp may be decided by the backend.
+  fixed64 start_time_unix_nano = 2;
+
+  // time_unix_nano is the moment when this aggregation value was reported.
+  //
+  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January
+  // 1970.
+  fixed64 time_unix_nano = 3;
+
+  // count is the number of values in the population. Must be non-negative. This
+  // value must be equal to the sum of the "count" fields in buckets if a
+  // histogram is provided.
+  fixed64 count = 4;
+
+  // sum of the values in the population. If count is zero then this field
+  // must be zero. This value must be equal to the sum of the "sum" fields in
+  // buckets if a histogram is provided.
+  double sum = 5;
+
+  // bucket_counts is an optional field contains the count values of histogram
+  // for each bucket.
+  //
+  // The sum of the bucket_counts must equal the value in the count field.
+  //
+  // The number of elements in bucket_counts array must be by one greater than
+  // the number of elements in explicit_bounds array.
+  repeated fixed64 bucket_counts = 6;
+
+  // A histogram may optionally contain the distribution of the values in the population.
+  // In that case one of the option fields below and "buckets" field both must be defined.
+  // Otherwise all option fields and "buckets" field must be omitted in which case the
+  // distribution of values in the histogram is unknown and only the total count and sum are known.
+
+  // explicit_bounds is the only supported bucket option currently.
+  // TODO: Add more bucket options.
+
+  // BucketUpperBound represents the bounding value of the histogram bucket such that all values
+  // in this bucket are less than or equal to (<=) the BucketUpperBound value.
+  //
+  // Motivation: We want to offer an HTTP/JSON endpoint for querying metrics.
+  // An "+Inf" bucket is implicitly added by prometheus client to histograms and,
+  // as JSON libraries cannot marshal +Inf bucket values, adding BucketUpperBound enables
+  // implementing the MarshalJSON function to stringify the "+Inf" bucket that is returned
+  // back from promQL queries.
+  message BucketUpperBound {
+    double upper_bound = 1;
+  }
```

Comment:
This was the only meaningful modification (addition) to the [OpenTelemetry metrics wire format](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto). 

Let me know if there is a better alternative to solving the JSON marshalling of `+Inf` values, or if I'm going down the wrong path somehow.